gradoop-common/src/test/java/org/gradoop/common/model/impl/properties/PropertyValueTest.java

```diff
@@ -1192,6 +1291,32 @@ public void testWriteAndReadFields() throws IOException {
     assertEquals(p, writeAndReadFields(PropertyValue.class, p));
   }
 
+  @Test
+  public void testReadAndWriteLargeString() throws Exception {
+    PropertyValue p = create(new String(new byte[PropertyValue.LARGE_PROPERTY_THRESHOLD]));
+    assertEquals(p, writeAndReadFields(PropertyValue.class, p));
+  }
+
+  @Test
+  public void testReadAndWriteLargeBigDecimal() throws Exception {
+    byte [] bigEndian = new byte[Short.MAX_VALUE + 10];
+    Arrays.fill(bigEndian, (byte) 121);
+    PropertyValue p = create(new BigDecimal(new BigInteger(bigEndian)));
+    assertEquals(p, writeAndReadFields(PropertyValue.class, p));
+  }
+
+  @Test
+  public void testReadAndWriteLargeMap() throws Exception {
+    HashMap<PropertyValue, PropertyValue> largeMap = new HashMap<>();
+    long neededEntries = PropertyValue.LARGE_PROPERTY_THRESHOLD;
```

Comment:
Although this will work, you can change it to a smaller value (like `(LARGE_PROPERTY_THRESHOLD / 10) +`), since keys and values are larger then 1 byte here (and theirs estimate size is known, since those are all strings of a certain length).